tests/ruby/misc_spec.rb

```diff
@@ -241,6 +241,18 @@
 
         expect(processes.primary.count_query("DISCARD ALL")).to eq(10)
       end
+
+      it "Resets server roles correctly" do
+        10.times do
+          conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user"))
+          conn.async_exec("SET SERVER ROLE to 'primary'")
+          conn.async_exec("SELECT 1")
+          conn.async_exec("SET statement_timeout to 5000")
+          conn.close
+        end
+
+        expect(processes.primary.count_query("RESET ROLE;")).to eq(10)
```

Comment:
can we `ALTER ROLE sharding_user ROLE TO <new role>` to simulate the same error we saw in production?

Reply:
Curious about the error, if you're willing to share. I'm not too familiar with `RESET USER`.

Reply:
@levkk we had 2 users in the DB, lets call them `A` and `B`.

We had run `ALTER ROLE A SET ROLE B`, which has the effect of running `SET ROLE B` after you open a connection with username `A`. We had configured `pgbouncer` and `pgcat` to both connect with username `A`.

When we run `create table` on pgbouncer, it was owned by `B` due to this config.

When we run `create table` on pgcat, a previous query had run `DISCARD ALL` in the pgcat server connection, and then the table was owned by `A`. We found that we could run `RESET ROLE` to fix it back to `SET ROLE b` after a `DISCARD ALL` was run.

Our permissions were all configured with assuming `B` owns the tables, so when we started creating tables with `A` as the owner after switching to pgcat, we had some errors.